beep-beep/jobs.md

```diff
@@ -0,0 +1,695 @@
+plugin. To enable it add `jobs` section to your configuration.
+
+For example, in this way, you can configure both the client and server parts to
+work with RabbitMQ.
+
+```yaml
+#
+# RPC is required for tasks dispatching (client)
+#
+rpc:
+  listen: tcp://127.0.0.1:6001
+
+#
+# This section configures the task consumer (server)
+#
+server:
+  command: php consumer.php
+  relay: pipes
+
+#
+# AMQP connection settings (for example, RabbitMQ)
+#
+amqp:
+    addr: amqp://guest:guest@localhost:5672
+
+#
+# In this section, the jobs themselves are configured
+#
+jobs:
+  consume: [ "test" ]   # List of RoadRunner queues that can be processed by 
+                        # the consumer specified in the "server" section.
+  pipelines:
+    test:               # RoadRunner queue identifier
+      driver: amqp      # - Queue driver name
+      queue: test       # - Internal (driver's) queue identifier
+```
+
+- The `rpc` section is responsible for client settings. It is at this address
+  that we will connect, *dispatching tasks* to the queue.
+
+- The `server` section is responsible for configuring the server. Previously, we
+  have already met with its description when setting up the [PHP Worker](/php/worker.md).
+
+- The `amqp` section contains settings for connecting to RabbitMQ. As the name
+  implies, the settings are specific only for the RabbitMQ driver and *may not*
+  be available if you use some other driver.
+
+- And finally, the `jobs` section is responsible for the work of the queues
+  themselves. It contains information on how the RoadRunner should work with
+  connections to drivers, what can be handled by the consumer, and other
+  queue-specific settings.
+
+### Common Configuration
+
+Let's now focus on the common settings of the queue server. In full, it may 
+look like this:
+
+```yaml
+jobs:
+  num_pollers: 64
+  pipeline_size: 100000
+  pool:
+    num_workers: 10
+    max_jobs: 0
+    allocate_timeout: 60s
+    destroy_timeout: 60s
+  consume: [ "queue-name" ]
+  pipelines:
+    queue-name:
+      driver: # "[DRIVER_NAME]"
+      # And driver-specific configuration below...
+```
+
+Above is a complete list of all possible common Jobs settings. Let's now figure
+out what they are responsible for.
+
+- `num_pollers` - TODO
+- `pipeline_size` - TODO
+- `pool.num_workers` - TODO
+- `pool.max_jobs` - TODO
+- `pool.allocate_timeout` - TODO
+- `pool.destroy_timeout` - TODO
+- `consume` - Contains an array of the names of all queues specified in the
+  `"pipelines"` section, which should be processed by the concierge specified in
+   the global `"server"` section (see the [PHP worker's settings](/php/worker.md)).
+- `pipelines` - This section contains a list of all queues declared in the
+  RoadRunner. The key is a unique *queue identifier*, and the value is an object
+  from the settings specific to each driver (we will talk about it later).
+
+### Ephemeral (Memory) Driver
+
+TODO
+
+### AMQP Driver
+
+TODO
+
```

Comment:
```json
      driver: amqp
      prefetch: 100
      queue: test-1-queue
      priority: 1
      exchange: default
      exchange_type: direct
      routing_key: test-1
      exclusive: false
      multiple_ack: false
      requeue_on_fail: false
```
Where: 
1. `driver` - driver's name.
2. `prefetch` - AMQP QoS, ref: https://www.rabbitmq.com/amqp-0-9-1-reference.html. This is the first arg of the qos method (`long prefetch-size`).
3. `queue` - AMQP queue name.
4. `priority` - pipeline priority. If the job has no priority, it will be inherited from this value.
5. `exchange_type` - direct, topics, headers, fanout.
6. `routing_key` - queue's routing key.
7. `exclusive` - exclusive queues can't be redeclared. If set to true and you'll try to declare the same pipeline twice, that will lead to an error.
8. `multiple_ack` - this delivery and all prior unacknowledged deliveries on the same channel will be acknowledged. This is useful for batch processing of deliveries. Applicable only for the Ack, not for the Nack.
9. `requeue_on_fail` - requeue on Nack.

Reply:
Looks like `prefetch` default is 100, isnt it? https://github.com/spiral/roadrunner/blob/feature/jobs_plugin/plugins/jobs/drivers/amqp/config.go#L50

Reply:
This is the development version. The default will be 10.